src/xmidtsend_rbus.c

```diff
@@ -327,14 +359,70 @@ void sendXmidtEventToServer(wrp_msg_t * msg)
 		if(msg_len > 0)
 		{
 			ParodusInfo("sendUpstreamMsgToServer\n");
-			sendUpstreamMsgToServer(&msg_bytes, msg_len);
+			sendRetStatus = sendUpstreamMsgToServer(&msg_bytes, msg_len);
 		}
+		else
+		{
+			ParodusInfo("The msg_len is zero\n");
+			xmidtQDequeue();
+			wrp_free_struct(notif_wrp_msg);
+			ParodusInfo("After wrp_free_struct\n");
+			free(msg_bytes);
+			msg_bytes = NULL;
+			return;
+		}
+
+		while(sendRetStatus)     //If SendMessage is failed condition
+		{
+			ParodusInfo("sendXmidtEventToServer is Failed\n");
+			if(highQosValueCheck(qos))
+			{
+				errorMsg = strdup("sendXmidtEventToServer Failed , Enqueue event since QOS is High");
+				ParodusInfo("The event is having high qos retry again \n");
+				//createOutParamsandSendAck(msg, asyncHandle, errorMsg, CLIENT_DISCONNECT);
+				ParodusInfo("The value of pointer xmidt->msg is %p\n", XmidtMsgQ->msg);
+				ParodusInfo("Wait till connection is Up\n");
+
+				pthread_cond_wait(&xmidt_con, &xmidt_mut);
```

Comment:
Don't use **xmidt_con** and **xmidt_mut** here, it should be a new cloud_status_cond and cloud_status_mut, as xmidt_con is getting triggered from producer whenever new requests are received and added to queue which is happening irrespective of cloud connection status.